### ISSUE TITLE
Create a "search" functionality

### ISSUE THREAD

**Issue by jakehamtexas** [the-pat/tokybook-dl#5]
Blocked by https://github.com/the-pat/tokybook-dl/issues/3

Implementation/API TBD.

**Comment by jakehamtexas** [comment]
So here's some things I've found:

1. You can search the site with the following URL + query param: `https://tokybook.com/?s=[string]`. 
2. It looks an awful lot like you can just URL encode the search string and it will Just Work. e.g. `https://tokybook.com/?s=hello+world` is what you get with the web app's search functionality, but it gives an equivalent result to `https://tokybook.com/?s=hello%20world`. 
3. Here's an example `curl` to see some output: `curl -A "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Brave Chrome/83.0.4103.116 Safari/537.36" https://tokybook.com/\?s\=hello+world`
4. It appears as if the resulting HTML has a reliable schema that we can use for grabbing information about the book. Here's a portion of the HTML you get from the example `curl` that has the relevant data for the search results: https://pastebin.com/99hA08Wk

Seems we could use an approach where we `GET` the search page, grab the article data per article, and display the GET urls back to the user for usage with the main command. It would be nice if we could use some degree of user-initiated pagination and selection in the CLI itself so that the experience is more integrated, but I think this might be a good first step. What do you think?

Speaking of pagination, this may be a separate issue, but here are some things I've found re: pagination:

1. It looks like the home page uses a `GET` to a url with the page number to buffer the next set of book items when the user scrolls down for an infinite scroll affect, e.g. `https://tokybook.com/page/3/`
2. Doing a `GET` from the browser to this URL is successful despite not having the previous pages data loaded.
3. There are `rel` links that reference the previous and next pages in the html, e.g.
![image](https://user-images.githubusercontent.com/31600776/116824404-9a095100-ab4f-11eb-9b5d-a5ca3636c833.png)
